ORStools/proc/directions_lines_proc.py

```diff
@@ -102,13 +104,18 @@ def processAlgorithm(self, parameters, context, feedback):
             context
         )
 
-        source_field_name = parameters[self.IN_FIELD]
-
-        (sink, dest_id) = self.parameterAsSink(parameters, self.OUT, context,
-                                               directions_core.get_fields(
-                                                   from_type=source.fields().field(source_field_name).type(),
-                                                   from_name=source_field_name,
-                                                   line=True),
+        # qgis overwrites PyQt5.QtCore.QVariant's __bool__ in
+        # https://github.com/qgis/QGIS/blob/master/python/PyQt/PyQt5/QtCore.py:
+        # def __bool__(self):
+        #     return not self.isNull()
+        if source_field_name := parameters[self.IN_FIELD]:
+            sink_fields = directions_core.get_fields(
+                    from_type=source.fields().field(source_field_name).type(),
+                    from_name=source_field_name, line=True)
+        else:
+            sink_fields = directions_core.get_fields(line=True)
+
+        (sink, dest_id) = self.parameterAsSink(parameters, self.OUT, context, sink_fields,
```

Comment:
maybe fill dynamically (no if-else, but only add options if)
```suggestion
        source_field_name = parameters[self.IN_FIELD]

        get_field_options = {
            "line": True
        }
        if source_field_name:
            get_field_options.update(
                from_type=source.fields().field(source_field_name).type(),
                from_name=source_field_name
            )

        sink_fields = directions_core.get_fields(**get_field_options)

        (sink, dest_id) = self.parameterAsSink(parameters, self.OUT, context, sink_fields,
```

Reply:
done - a bit differently though, as `line = True` is always set.